app/Http/Controllers/CampaignRegistrationController.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Campaign;
+use App\Models\User;
+use App\Models\Character;
+use Illuminate\Support\Facades\Auth;
+
+class CampaignRegistrationController extends Controller
+{
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'character_id' => "required|exists:characters,id",
+            'code' => "required|string"
+        ]);
+
+        $user = User::find(Auth::id())->first();
+        $character = Character::findOrFail($data['character_id'])->first();
+        $campaign = Campaign::where('code', '=', $data['code']);
```

Comment:
slightly irrelevant, but the operator parameter isn't really needed, so you can just do
```suggestion
        $campaign = Campaign::where('code', $data['code']);
```